src/app/services/sms/__tests__/sms.controller.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { ObjectId } from 'bson-ext'
+import _ from 'lodash'
+import { errAsync } from 'neverthrow'
+
+import expressHandler from 'tests/unit/backend/helpers/jest-express'
+
+import dbHandler from '../../../../../tests/unit/backend/helpers/jest-db'
+import { IFormSchema, IUserSchema } from '../../../../types'
+import { DatabaseError } from '../../../modules/core/core.errors'
+import * as FormService from '../../../modules/form/form.service'
+import * as SmsService from '../../../services/sms/sms.service'
+import * as SmsController from '../sms.controller'
+import { SmsCountsMeta } from '../sms.types'
+
+describe('sms.controller', () => {
+  let mockUser: IUserSchema
+  let mockForm: IFormSchema
+  const VERIFICATION_SMS_COUNT = 3
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  afterAll(async () => {
+    await dbHandler.clearDatabase()
```

Comment:
I don't think we should be calling the db layer in controller tests. Should we mock the db models instead?

Reply:
mocked the service layer a level below instead but yea, you're right! paraphrasing what you told me, even though we don't mock here, our tests are brittle because we are relying on behaviour that is external to this function for our tests. 

what should happen si that we mock the function's dependencies then see if the results are what we expect